Sources/PayPal/Buttons/PaymentButton.swift

```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+enum PayPalButtonType {
+    case payPal
+    case payPalCredit
+}
+
+/// Handles functionality shared across payment buttons
+public class PaymentButton: UIButton {
+
+    // asset identfier path for image and color button assets
+    private let bundleIdentifier = Bundle(identifier: "com.paypal.ios-sdk.PayPal")
+
+    // MARK: - Internal Helper Functions
+
+    func getButtonColor(for buttonType: PayPalButtonType) -> UIColor? {
+        switch buttonType {
+        case .payPal:
+            return UIColor(named: "PayPalGold", in: bundleIdentifier, compatibleWith: nil)
+        case .payPalCredit:
+            return UIColor(named: "PayPalDarkBlue", in: bundleIdentifier, compatibleWith: nil)
+        }
+    }
+
+    func getButtonLogo(for buttonType: PayPalButtonType) -> UIImage? {
+        switch buttonType {
+        case .payPal:
+            return UIImage(named: "PayPalLogo", in: bundleIdentifier, compatibleWith: nil)
+        case .payPalCredit:
+            return UIImage(named: "PayPalCreditLogo", in: bundleIdentifier, compatibleWith: nil)
+        }
+    }
+
+    func setLogoImageSize() -> UIEdgeInsets {
```

Comment:
maybe this can be change to `getLogoImageInsets`? since we just get but don't set the insets here